service-mesh/main.go

```diff
@@ -59,6 +59,11 @@ func run() (retErr error) {
 		return fmt.Errorf("failed to set up iptables rules: %w", err)
 	}
 
+	// Signal readiness for startup probe.
```

Comment:
We signal readiness, before we are actually ready here, could we use: https://www.envoyproxy.io/docs/envoy/latest/operations/admin.html#get--ready in the yaml as an HTTP probe?

Reply:
We don't want to expose the admin interface by default, do we? I don't think the probe works with localhost.

Reply:
Yeah exposing it locally might be fine, but it's generally added complexity.

Reply:
What we could do: 

1. Always expose the admin server on `127/8`.
2. Add an HTTP server to the proxy that forwards only `/ready`.